File: DIP/Spoilers/MarsRoverKata/Commands/FireMissileCommand.cs
namespace MarsRoverKata.Commands
{
    public class FireMissileCommand : ICommand
    {
        private Rover Rover { get; set; }

        public FireMissileCommand(Rover rover)
        {
            Rover = rover;
        }

        public bool Execute()
        {
            return Rover.FireMissle(); //Projectile(typeof(Missile));
        }
    }
}

File: DIP/Java_Exercise/MarsRoverKata/src/main/java/MarsRoverKata/Commands/FireMissileCommand.java
package MarsRoverKata.Commands;

import MarsRoverKata.Missile;
import MarsRoverKata.Rover;

public class FireMissileCommand implements ICommand {
    private Rover rover;

    public Rover getRover() {
        return rover;
    }

    public void setRover(Rover rover) {
        this.rover = rover;
    }

    public FireMissileCommand(Rover rover) {
        this.rover = rover;
    }

    public boolean execute() {
        return rover.fireMissile(); //Projectile(Missile.class);
    }
}

